imap/http_dblookup.c

```diff
@@ -276,12 +278,56 @@ static int get_uid2groups(struct transaction_t *txn,
     mboxlist_lookup(mboxname, &mbentry, NULL);
     if (!mbentry) goto done;
 
-    /* XXX init just incase carddav not enabled? */
+    /* Open DAV DB for the specified user.
+     * It will be assigned to an abook_set in jmap_get_accessible_addressbooks()
+     * and closed in jmap_free_abook_sets().
+     */
     db = carddav_open_userid(userid);
     if (!db) goto done;
 
-    array = carddav_getuid2groups(db, key, mbentry, otheruser);
-    if (!array) goto done;
+    if (!authstate) goto done;
```

Comment:
If we somehow get here with no authstate, we've already opened the dav db, and will fail to close it in `done`. This check should be almost the first thing this method does, probably before checking otherUser header

Reply:
Good catch, I moved the authstate check during testing.  I'll move it back above opening DAV DB.